native/dgeev.go

```diff
@@ -0,0 +1,283 @@
+	var wantvl bool
+	switch jobvl {
+	default:
+		panic("lapack: invalid JobLeftEV")
+	case lapack.ComputeLeftEV:
+		wantvl = true
+	case lapack.None:
+	}
+	var wantvr bool
+	switch jobvr {
+	default:
+		panic("lapack: invalid JobRightEV")
+	case lapack.ComputeRightEV:
+		wantvr = true
+	case lapack.None:
+	}
+	switch {
+	case n < 0:
+		panic(nLT0)
+	case len(work) < lwork:
+		panic(shortWork)
+	}
+	var minwrk int
+	if wantvl || wantvr {
+		minwrk = max(1, 4*n)
+	} else {
+		minwrk = max(1, 3*n)
+	}
+	if lwork != -1 {
+		checkMatrix(n, n, a, lda)
+		if wantvl {
+			checkMatrix(n, n, vl, ldvl)
+		}
+		if wantvr {
+			checkMatrix(n, n, vr, ldvr)
+		}
+		switch {
+		case len(wr) != n:
+			panic("lapack: bad length of wr")
+		case len(wi) != n:
+			panic("lapack: bad length of wi")
+		case lwork < minwrk:
+			panic(badWork)
+		}
+	}
+
+	// Quick return if possible.
+	if n == 0 {
+		work[0] = 1
+		return 0
+	}
+
+	maxwrk := 2*n + n*impl.Ilaenv(1, "DGEHRD", " ", n, 1, n, 0)
+	if wantvl || wantvr {
+		maxwrk = max(maxwrk, 2*n+(n-1)*impl.Ilaenv(1, "DORGHR", " ", n, 1, n, -1))
+		impl.Dhseqr(lapack.EigenvaluesAndSchur, lapack.UpdateZ, n, 0, n-1,
+			nil, 1, nil, nil, nil, 1, work, -1)
+		maxwrk = max(maxwrk, max(n+1, n+int(work[0])))
+		if wantvl {
+			impl.Dtrevc3(lapack.LeftEigVec, lapack.AllEigVecMulQ, nil, n,
+				nil, 1, nil, 1, nil, 1, n, work, -1)
+		} else {
+			impl.Dtrevc3(lapack.RightEigVec, lapack.AllEigVecMulQ, nil, n,
+				nil, 1, nil, 1, nil, 1, n, work, -1)
+		}
+		maxwrk = max(maxwrk, n+int(work[0]))
+		maxwrk = max(maxwrk, 4*n)
+	} else {
+		impl.Dhseqr(lapack.EigenvaluesOnly, lapack.None, n, 0, n-1,
+			nil, 1, nil, nil, nil, 1, work, -1)
+		maxwrk = max(maxwrk, max(n+1, n+int(work[0])))
+	}
+	maxwrk = max(maxwrk, minwrk)
+
+	if lwork == -1 {
+		work[0] = float64(maxwrk)
+		return 0
+	}
+
+	// Get machine constants.
+	smlnum := math.Sqrt(dlamchS) / dlamchP
+	bignum := 1 / smlnum
+
+	// Scale A if max element outside range [smlnum,bignum].
+	anrm := impl.Dlange(lapack.MaxAbs, n, n, a, lda, nil)
+	scalea := false
+	var cscale float64
+	if 0 < anrm && anrm < smlnum {
+		scalea = true
+		cscale = smlnum
+	} else if anrm > bignum {
+		scalea = true
+		cscale = bignum
+	}
+	if scalea {
+		impl.Dlascl(lapack.General, 0, 0, anrm, cscale, n, n, a, lda)
+	}
+
+	// Balance the matrix.
+	workbal := work[:n]
+	ilo, ihi := impl.Dgebal(lapack.PermuteScale, n, a, lda, workbal)
+
+	// Reduce to upper Hessenberg form.
+	iwrk := 2 * n
+	tau := work[n : iwrk-1]
+	impl.Dgehrd(n, ilo, ihi, a, lda, tau, work[iwrk:], lwork-iwrk)
+
+	var side lapack.EigVecSide
+	if wantvl {
+		side = lapack.LeftEigVec
+		// Copy Householder vectors to VL.
+		impl.Dlacpy(blas.Lower, n, n, a, lda, vl, ldvl)
+		// Generate orthogonal matrix in VL.
+		impl.Dorghr(n, ilo, ihi, vl, ldvl, tau, work[iwrk:], lwork-iwrk)
+		// Perform QR iteration, accumulating Schur vectors in VL.
+		iwrk = n
+		firstok = impl.Dhseqr(lapack.EigenvaluesAndSchur, lapack.UpdateZ, n, ilo, ihi,
```

Comment:
Make this firstOk or firstOK? I was reading it as "first tok" for most of the code and was really confused (yes, I realize that is nonsensical)